src/main/util.ts

```diff
@@ -295,37 +295,47 @@ const getSortedVideos = async (storageDir: string): Promise<VideoInfo[]> => {
  * more material than the user has allowed us.
  */
 const runSizeMonitor = async (storageDir: string, maxStorageGB: number): Promise<void> => {
+    let videoToDelete;
     const maxStorageBytes = maxStorageGB * Math.pow(1024, 3);
 
     let files = await getSortedVideos(storageDir);
     console.debug(`[Size Monitor] Running (max size = ${maxStorageGB} GB)`);
 
     files = files.map((file: any) => {
-        const metadata = getMetadataForVideo(file);
+        const metadata = getMetadataForVideo(file.name);
         return { ...file, metadata, };
     });
 
-    // Consider files with NO metadata (dangling video files)
-    // and files that ARE NOT protected
-    files = files.filter((file: any) => !file.hasOwnProperty('metadata') || !Boolean(file.metadata.protected));
+    // Files without metadata are considered dangling and are cleaned up. 
+    const danglingFiles = files.filter((file: any) => !file.hasOwnProperty('metadata') || !file.metadata);
+    const unprotectedFiles = files.filter((file: any) => file.hasOwnProperty('metadata') && file.metadata && !Boolean(file.metadata.protected));
+
+    if (danglingFiles.length !== 0) {
```

Comment:
I'm thinking to make a list of "videosToDelete" instead and add to that, both those over size and also those dangling ones and _then_ delete all those instead of having two sections with deleteVideo?

Unless you want the log to reflect what is being done in detail

Reply:
Probably would be a bit neater but just going to stick with this.